stdlib/pathlib.pyi

```diff
@@ -198,6 +200,10 @@ class Path(PurePath):
         def write_text(self, data: str, encoding: str | None = ..., errors: str | None = ...) -> int: ...
     if sys.version_info >= (3, 8):
         def link_to(self, target: StrPath | bytes) -> None: ...
+    if sys.version_info >= (3, 12):
+        def walk(
+            self: Self, top_down: bool = ..., on_error: _OnError | None = ..., follow_symlinks: bool = ...
```

Comment:
The `_OnError` `TypeAlias` is only used once in this file (unlike in our stubs for `os`, where the alias is reused multiple times in the file). I don't think there's any reason to define it at all here in `pathlib.pyi`; let's just use `Callable[[OSError], object]` directly:

```suggestion
            self: Self, top_down: bool = ..., on_error: Callable[[OSError], object] | None = ..., follow_symlinks: bool = ...
```